### PR TITLE
Add `Data.Map` Instance to `Metadata.FromMetadata`

### PR DESCRIPTION
**Pull request by MitchyCola**
- This adds instances for `Data.Map` in `Metadata.FromMetadata`, ~~`ToData`, and `FromData`~~. 

- This also fixes the `Map` instance for `Metadata.ToMetadata`, where the name of the exported variable was causing errors in a bundled `output.js`.

### Pre-review checklist

- [x] All code has been formatted using our config (`make format` for Purescript, `nixpkgs-fmt` for Nix)
- [x] All Purescript imports are explicit, including constructors
- [x] **All** existing examples have been run locally against a fully-synced testnet node
- [x] The integration and unit tests have been run (`npm run test`) locally
- [x] The changelog has been updated under the `## Unreleased` header, using the appropriate sub-headings (`### Added`, `### Removed`, `### Fixed`)

### CODE REVIEW

**Comment by klntsky** [comment]
Plutus data uses a map that is backed by a kv-list. I am not sure at all these instances are correct (because of different handling of  duplicates). Could you please look into Plutus code, how is it handled there?

> the name of the exported variable

Which name?

**Comment by vvtran** [comment]
I agree with @klntsky, we shouldn't be using `Data.Map.Map`, instead we should probably use `Plutus.Types.AssocMap.Map`

**Comment by vvtran** [comment]
Unless there is a specific reason to use PureScript's Map?

**Comment by MitchyCola** [comment]
> > the name of the exported variable
> 
> Which name?

@klntsky Previously, when I was trying to bundle while using `toMetadata` on a Map, I was getting this error:

```shell
ERROR in ./output.js 11088:10
Module parse failed: Unterminated string constant (11088:10)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|   exports["toMetadataMap"] = toMetadataMap;
|   exports["toMetadataArray/\AnyToMet"] = toMetadataArray$div$bslashAnyToMet;
>   exports["toMetadataArray/\"] = toMetadataArray$div$bslash;
|   exports["toMetadataByteArray"] = toMetadataByteArray;
|   exports["toMetadataString"] = toMetadataString;
 @ ./main.js 8:0-21

webpack 5.67.0 compiled with 1 error in 610 ms
```

The backshlasses from the tuple symbol, `/\`, were being used to escape the quotation marks. So I swapped them out for `Tuple` in the instance declaration for `ToMetadata (Array (k /\ AnyToMetadata))` and `ToMetadata (Array (k /\ v))`, in `Metadata.ToMetadata`.

**Comment by MitchyCola** [comment]
I was using `Data.Map` because `Metadata.ToMetadata` was already using it in its instances. I can look into using `Plutus.Types.AssocMap`.

**Comment by MitchyCola** [comment]
I removed `Data.Map.Map` instances for `FromData` and `ToData`, there are instances for them in `Plutus.Types.AssocMap.Map`.

`Data.Map.Map` should be kept for `Metadata.FromMetadata` and `Metadata.ToMetadata`.

**Comment by MitchyCola** [comment]
@vvtran  The simple answer would be that the type class definition for `Types.TransactionMetadata.TransactionMetadatum` uses `Data.Map.Map`. Which is also used in `Deserilization.Transaction`, 'Metadata.Helpers`, & `Serialization.AuxilaryData`

I suppose that it could be changed to `Plutus.Types.AssocMap.Map` to keep consistency. If it were to be changed, it would probably be better to do it in another PR.

File: CHANGELOG.md [hunk 1]
```diff
@@ -49,6 +50,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - `ConfigParams` is now a type synonym instead of a newtype. `ContractConfig` has been renamed to `ContractEnv`.
 - Moved logging functions to `Contract.Log` from `Contract.Monad` ([#727](https://github.com/Plutonomicon/cardano-transaction-lib/issues/727)
 - Renamed `Contract.Wallet.mkKeyWalletFromPrivateKey` to `Contract.Wallet.mkKeyWalletFromPrivateKeys`.
+- Rplaced `/\` with `Tuple` for the `Array Tuple` instances in `Metadata.ToMetadata` ([#633](https://github.com/Plutonomicon/cardano-transaction-lib/pull/633))
```
Comment (klntsky) [inline CHANGELOG.md:53]:
```suggestion
- Replaced `/\` with `Tuple` for the `Array Tuple` instances in `Metadata.ToMetadata` ([#633](https://github.com/Plutonomicon/cardano-transaction-lib/pull/633))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -39,6 +39,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - `Contract.Wallet.isNamiAvailable` and `Contract.Wallet.isGeroAvailable` functions ([#558](https://github.com/Plutonomicon/cardano-transaction-lib/issues/558)])
 - `Contract.Transaction.balanceTxWithOwnAddress` and `Contract.Transaction.balanceTxsWithOwnAddress` to override an `Address` used in `balanceTx` internally ([#775](https://github.com/Plutonomicon/cardano-transaction-lib/pull/775))
 - `Contract.Transaction.awaitTxConfirmedWithTimeoutSlots` waits a specified number of slots for a transaction to succeed. ([#790](https://github.com/Plutonomicon/cardano-transaction-lib/pull/790))
+- `Data.Map.Map` instance for `Metadata.FromMetadata` ([#633](https://github.com/Plutonomicon/cardano-transaction-lib/pull/633))
```
Comment (klntsky) [inline CHANGELOG.md:42]:
```suggestion
- `Metadata.FromMetadata` instance for `Data.Map.Map` ([#633](https://github.com/Plutonomicon/cardano-transaction-lib/pull/633))
```

**Review by klntsky — CHANGES_REQUESTED** [review]

**Comment by ngua** [comment]
There hasn't been much activity on this recently so I'm going to close it for now. If the instance is still required please feel free to reopen!

**Pull request closed without merging**